src/ngx_http_lua_args.c

```diff
@@ -40,10 +73,14 @@ ngx_http_lua_ngx_req_set_uri_args(lua_State *L)
 
     ngx_http_lua_check_fake_request(L, r);
 
-    switch (lua_type(L, 1)) {
+    type = lua_type(L, 1);
+    switch (type) {
     case LUA_TNUMBER:
     case LUA_TSTRING:
         p = (u_char *) lua_tolstring(L, 1, &len);
+        if (type == LUA_TSTRING && !ngx_http_lua_valid_args(p, len)) {
```

Comment:
I suggest you no longer combine the `LUA_TNUMBER` and `LUA_TSTRING` cases together otherwise you are testing against the `LUA_TSTRING` twice here...